lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -184,6 +191,16 @@ pub fn generate_constraint_mut(f: &Field, c: &ConstraintMut) -> proc_macro2::Tok
     }
 }
 
+pub fn generate_constraint_dup(f: &Field, c: &ConstraintDup) -> proc_macro2::TokenStream {
+    let me = &f.ident;
+    let target = &c.target;
+    quote! {
+        if !(#me.to_account_info().key == #target.to_account_info().key) {
```

Comment:
I was imaging the `dup`  constraint to be less strict. That is, it would be more accurate to say *maybe* dup. And so instead of adding this assertion, it would simply skip being included from the dup checks below. I don't think there's a great reason to always expect the client to provide a duplicate account, since you can always just clone from within the program.

Reply:
yea that makes more sense. if you have two accounts and you want to set them to always be duplicates, you probably dont need two accounts in the first place.